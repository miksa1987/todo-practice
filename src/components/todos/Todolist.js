import React from 'react'
import Todo from './Todo'

const Todolist = (props) => {
  return (
    <div>
      {props.todos.map((todo, index) => 
        <Todo 
          key={index} 
          todo={todo}
          toggleDone={props.toggleDone} 
          removeTodo={props.removeTodo}
        />)}
    </div>
  )
}

export default Todolist