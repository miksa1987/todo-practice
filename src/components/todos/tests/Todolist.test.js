import { render, cleanup } from '@testing-library/react'
import React from 'react'
import Todolist from '../Todolist'

const testTodos = [
  { text: 'Tee tämä appi valmiiksi', done: false, id: 0 },
  { text: 'Ja myös testit!', done: false, id: 1 }
]

describe('Todolist', () => {
  afterEach(cleanup)

  test('Renders correctly', () => {
    const component = render(<Todolist todos={testTodos} />)

    expect(component).toMatchSnapshot()
  })
})