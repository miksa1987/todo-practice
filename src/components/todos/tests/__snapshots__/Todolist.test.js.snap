// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todolist Renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-htpNat hIjJNZ"
        >
          <button
            id="todo-toggle"
          >
            <svg
              class="sc-bwzfXH krgfXf"
              id="todo-active"
            >
              cross.svg
            </svg>
          </button>
          <p>
            Tee tämä appi valmiiksi
          </p>
          <button
            id="todo-remove"
          >
            X
          </button>
        </div>
        <div
          class="sc-htpNat hIjJNZ"
        >
          <button
            id="todo-toggle"
          >
            <svg
              class="sc-bwzfXH krgfXf"
              id="todo-active"
            >
              cross.svg
            </svg>
          </button>
          <p>
            Ja myös testit!
          </p>
          <button
            id="todo-remove"
          >
            X
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-htpNat hIjJNZ"
      >
        <button
          id="todo-toggle"
        >
          <svg
            class="sc-bwzfXH krgfXf"
            id="todo-active"
          >
            cross.svg
          </svg>
        </button>
        <p>
          Tee tämä appi valmiiksi
        </p>
        <button
          id="todo-remove"
        >
          X
        </button>
      </div>
      <div
        class="sc-htpNat hIjJNZ"
      >
        <button
          id="todo-toggle"
        >
          <svg
            class="sc-bwzfXH krgfXf"
            id="todo-active"
          >
            cross.svg
          </svg>
        </button>
        <p>
          Ja myös testit!
        </p>
        <button
          id="todo-remove"
        >
          X
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
